### PR TITLE
Fix tests using dml and fix notifier in dml mover

### PR DESCRIPTION
**Pull request by blazej-smorawski**
Tests using dml were  using the wrong path for check_movdir64b and they were always passing because of that. Wrong usage of notifier in dml mover would cause a segfault if NULL notifier was passed.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/miniasync/73)

### CODE REVIEW

**Comment by kswiecicki** [comment]
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 17 at r1](https://reviewable.io/reviews/pmem/miniasync/73#-MyqBgw89_dzsJSCZlRX:-MyqBgwH-GnOTtaApusy:bt1cyw0) ([raw file](https://github.com/pmem/miniasync/blob/0545a1d257e51dd2367d9a65bd6662c12633ed60/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L17)):*
> ```CMake
> else()
>     message("Wrong return from check_movdir64b:")
>     message(${RET})
> ```

You use `${RET}` in the message above.



_Code quote:_
```CMake
message(${RET})
```

**Review by blazej-smorawski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/73)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @kswiecicki)

___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 17 at r1](https://reviewable.io/reviews/pmem/miniasync/73#-MyqBgw89_dzsJSCZlRX:-MyqCGag1Wyxf6bhu5P3:b-1akba5) ([raw file](https://github.com/pmem/miniasync/blob/0545a1d257e51dd2367d9a65bd6662c12633ed60/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L17)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

You use `${RET}` in the message above.
</blockquote></details>

What's wrong with that? I wanted to print a message when something goes wrong, so in the previous case when it wasn't working properly we would get "No such a file etc."

**Comment by kswiecicki** [comment]
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 17 at r1](https://reviewable.io/reviews/pmem/miniasync/73#-MyqBgw89_dzsJSCZlRX:-MyqCaFg9pVxI0HJtH7k:biw39nj) ([raw file](https://github.com/pmem/miniasync/blob/0545a1d257e51dd2367d9a65bd6662c12633ed60/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L17)):*
<details><summary><i>Previously, blazej-smorawski wrote…</i></summary><blockquote>

What's wrong with that? I wanted to print a message when something goes wrong, so in the previous case when it wasn't working properly we would get "No such a file etc."
</blockquote></details>

Sry, I meant to say 'You can use ...". I'm wondering why not just go with `message("Wrong return from check_movdir64b: ${RET}")`

**Comment by kswiecicki** [comment]
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 15 at r2](https://reviewable.io/reviews/pmem/miniasync/73#-MyqEN5W3SkOoCWaTkTB:-MyqEN5XAki8x099Gtvf:bf32pls) ([raw file](https://github.com/pmem/miniasync/blob/1e325ba4df393a5470511d33839153e1e1245189/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L15)):*
> ```CMake
> if (RET EQUAL 0)
>     execute(0 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${BUILD}/vdm_operation_future_poll)
> else()
> ```

You should add `elseif(RET EQUAL -1)` to check for when the return value is not wrong, but the movdir64b instruction is not set.



_Code quote:_
```CMake
else()
```

File: tests/vdm_operation_future_poll/vdm_operation_future_poll.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	if (ret != 0) {
+		return ret;
+	}
+
+	struct data_mover_dml *dmd = data_mover_dml_new(DATA_MOVER_DML_AUTO);
+	ret = test_vdm_future_poll(data_mover_dml_get_vdm(dmd), size);
+	data_mover_dml_delete(dmd);
+	if (ret != 0) {
+		return ret;
+	}
```
Comment (DamianDuy) [inline tests/vdm_operation_future_poll/vdm_operation_future_poll.c:53]: It doesn't seem necessary to check this, since `ret` is returned later anyway.

**Review by pbalcer — CHANGES_REQUESTED** [review]
> Reviewed 3 of 5 files at r1, 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/73)* status: 4 of 5 files reviewed, 4 unresolved discussions (waiting on @blazej-smorawski, @DamianDuy, @kswiecicki, and @pbalcer)

___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 9 at r2](https://reviewable.io/reviews/pmem/miniasync/73#-MyqxcqwAXu1k8WzEj8d:-MyqxcqwAXu1k8WzEj8e:b4m7ejq) ([raw file](https://github.com/pmem/miniasync/blob/1e325ba4df393a5470511d33839153e1e1245189/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L9)):*
> ```CMake
> 
> # check for MOVDIR64B instruction
> execute_process(COMMAND ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${BUILD}/check_movdir64b
> ```

you could probably implement a generic function that does this. would look nicer.

**Review by blazej-smorawski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/73)* status: 2 of 7 files reviewed, 6 unresolved discussions (waiting on @DamianDuy, @kswiecicki, and @pbalcer)

___
*[extras/dml/utils/check_movdir64b.c, line 10 at r3](https://reviewable.io/reviews/pmem/miniasync/73#-Myv7Wyw3qSrUovZ24Ks:-Myv7WyxDui9UDWkj8Jt:b29gsor) ([raw file](https://github.com/pmem/miniasync/blob/cd571ae756ad302173ec86995622ff43108a4486/extras/dml/utils/check_movdir64b.c#L10)):*
> ```C
> {
> 	if (!is_cpu_movdir64b_present())
> 		return 1;
> ```

I changed it to 1, because when I tested these return codes on my machines returning -1 resulted in 255. Returning 1 is safer no matter the number encoding.
___
*[tests/cmake/test_helpers.cmake, line 171 at r3](https://reviewable.io/reviews/pmem/miniasync/73#-Myv7umO6o8tt2mT-5NU:-Myv7umO6o8tt2mT-5NV:b-9aexeu) ([raw file](https://github.com/pmem/miniasync/blob/cd571ae756ad302173ec86995622ff43108a4486/tests/cmake/test_helpers.cmake#L171)):*
> ```CMake
> 	else()
> 		set(MOVDIR64B -1 PARENT_SCOPE)
> 		message(FATAL_ERROR "Error while testing for movdir64b. Error message: ${RET}")
> ```

If running check_movdir64b returns unknown code or something like "No such a file..." the tests using it will now fail properly instead of passing the test as it happened before.
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 17 at r1](https://reviewable.io/reviews/pmem/miniasync/73#-MyqBgw89_dzsJSCZlRX:-MyqCwtVCBWzAJHvvm8x:bb74ce2) ([raw file](https://github.com/pmem/miniasync/blob/0545a1d257e51dd2367d9a65bd6662c12633ed60/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L17)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

Sry, I meant to say 'You can use ...". I'm wondering why not just go with `message("Wrong return from check_movdir64b: ${RET}")`
</blockquote></details>

Sure
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 9 at r2](https://reviewable.io/reviews/pmem/miniasync/73#-MyqxcqwAXu1k8WzEj8d:-Myv4trO-bUwglc9LMXm:bxh7flt) ([raw file](https://github.com/pmem/miniasync/blob/1e325ba4df393a5470511d33839153e1e1245189/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L9)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

you could probably implement a generic function that does this. would look nicer.
</blockquote></details>

Sure, now it's much better and the function also throws `FATAL_ERROR` when the check couldn't be run properly.
___
*[tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake, line 15 at r2](https://reviewable.io/reviews/pmem/miniasync/73#-MyqEN5W3SkOoCWaTkTB:-Myv4qFLEaBCPOL69lp7:b-4rjpze) ([raw file](https://github.com/pmem/miniasync/blob/1e325ba4df393a5470511d33839153e1e1245189/tests/vdm_operation_future_poll/test_vdm_operation_future_poll.cmake#L15)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

You should add `elseif(RET EQUAL -1)` to check for when the return value is not wrong, but the movdir64b instruction is not set.
</blockquote></details>

Sure, now it's all inside function `check_movdir64b`.
___
*[tests/vdm_operation_future_poll/vdm_operation_future_poll.c, line 53 at r2](https://reviewable.io/reviews/pmem/miniasync/73#-MyqA7d7k_lQ2MfTO6Dc-r2-53:-MyusmST7GqBATl3M0dz:b-l5injb) ([raw file](https://github.com/pmem/miniasync/blob/1e325ba4df393a5470511d33839153e1e1245189/tests/vdm_operation_future_poll/vdm_operation_future_poll.c#L53)):*
<details><summary><i>Previously, DamianDuy (Damian Duy) wrote…</i></summary><blockquote>

It doesn't seem necessary to check this, since `ret` is returned later anyway.
</blockquote></details>

Actually, I removed them all, because there won't be a different return value than 0 and the test is more about looking for segfaults etc. Maybe if there will be some kind of notifier error that can be returned while polling, we can add something like that.

**Comment by kswiecicki** [comment]
___
*[extras/dml/utils/check_movdir64b.c, line 10 at r3](https://reviewable.io/reviews/pmem/miniasync/73#-Myv7Wyw3qSrUovZ24Ks:-MyvOJ7ECuq2ZsxKwPwQ:b-80tlqk) ([raw file](https://github.com/pmem/miniasync/blob/cd571ae756ad302173ec86995622ff43108a4486/extras/dml/utils/check_movdir64b.c#L10)):*
<details><summary><i>Previously, blazej-smorawski wrote…</i></summary><blockquote>

I changed it to 1, because when I tested these return codes on my machines returning -1 resulted in 255. Returning 1 is safer no matter the number encoding.
</blockquote></details>

Sure.

**Review by DamianDuy — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/73#-:-MyvxVCd856dSEkTWLym:bnfp4nl)* status: all files reviewed, 5 unresolved discussions (waiting on @blazej-smorawski, @DamianDuy, @kswiecicki, and @pbalcer)

**Pull request merged**